### PR TITLE
Feature/rm last read provider

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Move last LastReadBlockProvider to notary-commons

### CODE REVIEW

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 1]
```diff
@@ -78,9 +79,9 @@ class ChainAdapter(
         return Result.of {
             if (chainAdapterConfig.dropLastReadBlock) {
                 logger.info { "Drop last block" }
-                lastReadBlockProvider.saveLastBlockHeight(0)
+                lastReadBlockProvider.saveLastBlockHeight(BigInteger.ZERO)
             }
-            lastReadBlock.set(lastReadBlockProvider.getLastBlockHeight())
+            lastReadBlock.set(lastReadBlockProvider.getLastBlockHeight().toLong())
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: I suggest you use BigInteger without casting to long.

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 2]
```diff
@@ -45,12 +45,18 @@ class ChainAdapter(
...
 
     private val connectionFactory = ConnectionFactory()
-    private val lastReadBlock = AtomicLong()
+
     private val publishUnreadLatch = CountDownLatch(1)
     private val subscriberExecutorService = createPrettySingleThreadPool(
         CHAIN_ADAPTER_SERVICE_NAME, "iroha-chain-subscriber"
     )
 
+    var lastReadBlock: BigInteger = BigInteger.ZERO
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: I'd rather use `AtomicReference`.

Comment (Alexey-N-Chernyshov) [reply src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: As you wish.

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 3]
```diff
@@ -45,12 +46,14 @@ class ChainAdapter(
...
 
     private val connectionFactory = ConnectionFactory()
-    private val lastReadBlock = AtomicLong()
+
     private val publishUnreadLatch = CountDownLatch(1)
     private val subscriberExecutorService = createPrettySingleThreadPool(
         CHAIN_ADAPTER_SERVICE_NAME, "iroha-chain-subscriber"
     )
 
+    private var lastReadBlock = AtomicReference<BigInteger>()
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: I'd rather use `val`. I also suggest you set BigInteger.ZERO as a default value.

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 4]
```diff
@@ -101,7 +107,7 @@ class ChainAdapter(
                     .subscribe({ block ->
                         publishUnreadLatch.await()
                         // Send only not read Iroha blocks
-                        if (block.blockV1.payload.height > lastReadBlock.get()) {
+                        if (block.blockV1.payload.height > lastReadBlock.get().toLong()) {
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: Given the fact that `lastReadBlock.get()` returns `BigInteger`, we may go out of positive integers. I'd rather cast `block.blockV1.payload.heigh` instead.

File: src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt [hunk 5]
```diff
@@ -116,7 +122,7 @@ class ChainAdapter(
      * @param channel - RabbitMQ channel that is used to publish Iroha blocks
      */
     private fun publishUnreadIrohaBlocks(channel: Channel) {
-        var lastProcessedBlock = lastReadBlockProvider.getLastBlockHeight()
+        var lastProcessedBlock = lastReadBlockProvider.getLastBlockHeight().toLong()
```
Comment (dolgopolovwork) [inline src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt]: The same problem here.

**Pull request merged**